src/main/java/core/tasks/Deadline.java

```diff
@@ -85,7 +86,8 @@ public static Deadline getInstance(String description, String by) throws NoDeadl
     @Override
     public String toString() {
         OutputFormatter outputFormatter = OutputFormatter.getInstance();
-        outputFormatter.appendAll("[D]", super.toString(), " (by: ", byDate.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
+        outputFormatter.appendAll("[D]", super.toString(), " (by: ",
+                byDate.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
         if (byTime != null) {
```

Comment:
Perhaps there could be another way of handling this check other than using a null check? 🤔